packages/jsii-rosetta/lib/commands/extract.ts

```diff
@@ -50,50 +40,44 @@ export async function extractSnippets(
 
   logging.info(`Loading ${assemblyLocations.length} assemblies`);
   const assemblies = await loadAssemblies(assemblyLocations, options.validateAssemblies);
-  const fingerprinter = new TypeFingerprinter(assemblies.map((a) => a.assembly));
 
   let snippets = Array.from(allTypeScriptSnippets(assemblies, loose));
   if (only.length > 0) {
     snippets = filterSnippets(snippets, only);
   }
 
-  const tablet = new LanguageTablet();
+  const translator = (options.translatorFactory ?? mk(RosettaTranslator))({
+    assemblies: assemblies.map((a) => a.assembly),
+  });
 
   if (options.cacheTabletFile) {
-    await reuseTranslationsFromCache(snippets, tablet, options.cacheTabletFile, fingerprinter);
+    await translator.loadCache(options.cacheTabletFile);
+    const reused = translator.readFromCache(snippets);
```

Comment:
Lets put a comment here noting that this removes/manipulates `snippets`. It's probably documented on the method itself, but reading this flow as-is it's certainly not clear.